libs/math/include/math/TVecHelpers.h

```diff
@@ -497,6 +497,14 @@ class TVecFunctions {
         return v;
     }
 
+    friend inline constexpr VECTOR<T> MATH_PURE mix(VECTOR<T> x, VECTOR<T> y, T a) noexcept {
+        return x * (VECTOR<T>(1) - a) + y * a;
+    }
+
+    friend inline constexpr VECTOR<T> MATH_PURE lerp(VECTOR<T> x, VECTOR<T> y, T a) noexcept {
```

Comment:
`a` could be a vector too. Implicat constructors should make using scalar constants doable too this way.

Reply:
You're right, GLSL supports this so we should too.  Weird.

Reply:
The existing mix/lerp in scalar.h already handles the `vector,vector,vector` overload, but I went ahead and added a special overload anyway since not everybody includes scalar.h.

I had to use `VECTOR<T>, VECTOR<T>, VECTOR<T>` instead of `VECTOR<T>, VECTOR<T>, T`.  The latter would have been clever and terse, but causes an ambiguity error.

Reply:
`VECTOR<T>, VECTOR<T>, VECTOR<T>` is what I was interested in :)

Reply:
Do you approve of the PR in its current state?